File: python/examples/producer.py
import qbus
import signal
import sys

if len(sys.argv) < 4:
    print "Usage: python producer.py config_path topic_name cluster_name"
    sys.exit(1)

config_path = sys.argv[1]
topic_name = sys.argv[2]
cluster_name = sys.argv[3]

producer = qbus.QbusProducer()
if False == producer.init(cluster_name, "producer.log", config_path, topic_name):
    print "Init failed"
    sys.exit(2)

print "%% Please input messages (Press Ctrl+C or Ctrl+D to exit):"

while True:
    try:
        line = raw_input()
        if False == producer.produce(line, len(line), ""):
            print "Failed to produce"
    except (EOFError, KeyboardInterrupt):
        producer.uninit()
        print "%% Done."
        break

File: php/examples/producer.php
<?php

declare(ticks = 1);

require("./qbus.php");

if (count($argv) < 4)
{
    echo "Usage: php producer.php config_path topic_name cluster_name\n";
    exit(1);
}

$config = $argv[1];
$topic = $argv[2];
$cluster = $argv[3];

$producer = new QbusProducer;
if (false == $producer->init($cluster, "./producer.log", $config, $topic))
{
    echo "Init failed\n";
    exit(1);
}

echo "%% Please input messages (Press Ctrl+D to exit):\n";

while (true)
{
    $line = fgets(STDIN);
    if (feof(STDIN)) {
        break;
    }

    $line = str_replace(array("\r", "\n"), '', $line);
    if (false == $producer->produce($line, strlen($line), ""))
    {
        echo "Failed to produce\n";
    }
}

$producer->uninit();
echo "%% Done.\n";

?>
